configurator/sshkey/sshkey.go

```diff
@@ -0,0 +1,111 @@
+package sshkey
+
+import (
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/user"
+	"path"
+	"path/filepath"
+	"strconv"
+
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/sys/unix"
+
+	"github.com/percona/pmm-manage/configurator/config"
+)
+
+func Init(c config.PMMConfig) Handler {
+	return Handler{
+		KeyPath:  c.SSHKeyPath,
+		KeyOwner: c.SSHKeyOwner,
+	}
+}
+
+func (c *Handler) RunSSHKeyChecks() {
+	sshKeyUser, err := user.Lookup(c.KeyOwner)
+	if err != nil {
+		log.Fatal(err)
+	}
+	if c.KeyPath == "" {
+		c.KeyPath = path.Join(sshKeyUser.HomeDir, ".ssh/authorized_keys")
+	}
+
+	sshKeyDir := filepath.Dir(c.KeyPath)
+	logger := log.WithField("dir", sshKeyDir)
+	if dir, err := os.Stat(sshKeyDir); err != nil || !dir.IsDir() {
+		if err := os.MkdirAll(sshKeyDir+"/", 0700); err != nil {
+			logger.WithField("error", err).Fatal("Cannot create ssh directory")
+		}
+		uid, _ := strconv.Atoi(sshKeyUser.Uid)
+		gid, _ := strconv.Atoi(sshKeyUser.Gid)
+		if err := os.Chown(sshKeyDir, uid, gid); err != nil {
+			logger.WithField("error", err).Fatal("Cannot change owner of ssh directory")
+		}
+	}
+	if err := unix.Access(sshKeyDir, unix.W_OK); err != nil {
```

Comment:
fixed in https://github.com/percona/pmm-manage/pull/9/commits/c0ae429e079e9a55606c3762443fd05f62fdee01